### DISCUSSION TITLE
template x-for, how to use the "key" keyword

### DISCUSSION THREAD

**Discussion by ericxin1982** [alpinejs/alpine#3630] [1. Help] [answered]
<ul x-data="{ colors: ['Red', 'Orange', 'Yellow'] }">
    <template x-for="(color, index) in colors" :key="index">
        <li>
            <span x-text="index + ': '"></span>
            <span x-text="key + ': '"></span>
            <span x-text="color"></span>
        </li>
    </template>
</ul>

Hi Team,

I am learning alpine, and see template x-for, which has key this keyword, and I try to explore the value from loop, but it failed, it said key is not defined, how to figure this out?

Thanks
Eric Xin

**Comment by ericxin1982** [comment]
![image](https://github.com/alpinejs/alpine/assets/9047436/08eb3e25-9590-4c97-856b-3fb489b7a86d)

**Comment by ekwoka** [comment, marked as answer]
The key isn't available in the state.

It's only used internally by the `x-for` directive.

You don't access it, it is simply used to keep track of which elements are what during looping.

**Reply by ericxin1982** [reply]
ok
